spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/spanner/SpannerTransactionManagerAutoConfiguration.java

```diff
@@ -39,9 +40,11 @@
 @Configuration
 @EnableTransactionManagement
 @ConditionalOnClass(SpannerTransactionManager.class)
-@AutoConfigureOrder(Ordered.LOWEST_PRECEDENCE)
+@AutoConfigureBefore(PlatformTransactionManager.class)
+@ConditionalOnProperty(value = "spring.cloud.gcp.spanner.transactional-annotation.enabled",
```

Comment:
`spring.cloud.gcp.spanner.transactions.enabled` because you can use the transaction manager without `@Transactional`?

Reply:
Yes the `@Transactional` is the only one affected. The modern style transactions-in-a-function is still supported.

Reply:
Why can't we live without this property? The fact is that no any other transaction management auto-configuration does something similar...

Reply:
I'm OK with it being on permanently actually. @meltsufin WDYT

Reply:
The reason we need it, I believe, is because we can't control it via classpath. It was Mark's suggestion, and I agree with it. We can, of course, decide to enable it by default.

Reply:
Well, not sure. We don't control it in case of JDBC TX Manager either. There is just enough to have a `spring-jdbc` in classpath and that's all.
We don't do that for the mentioned `Neo4jTransactionManager`, too, nor for `JpaTransactionManager`.
But if insist, let it be. We can always remove it afterward.

Only the problem that this has to be included into some `@ConfigurationProperties` or added into the `additional-spring-configuration-metadata.json`.

Reply:
Sure, if no other Spring Data project does it, I'm fine with not doing too.

Reply:
I actually didn't know about `additional-spring-configuration-metadata.json` until just now. What is the significance of this file, and how come only a few options are listed in there?

Reply:
removed.

Reply:
> I actually didn't know about additional-spring-configuration-metadata.json until just now.

See this issue: https://github.com/spring-cloud/spring-cloud-gcp/issues/973